spot_ros2_control/hardware/include/spot_ros2_control/spot_joint_map.hpp

```diff
@@ -0,0 +1,109 @@
+/// @brief Number of joints we expect if the robot has no arm
+static const int kNjointsNoArm = 12;
+
+/// @brief Maps joint name to desired joint index for robots with arms
+static const std::unordered_map<std::string, size_t> kJointNameToIndexWithArm{
+    {"front_left_hip_x", 0},  {"front_left_hip_y", 1}, {"front_left_knee", 2},   {"front_right_hip_x", 3},
+    {"front_right_hip_y", 4}, {"front_right_knee", 5}, {"rear_left_hip_x", 6},   {"rear_left_hip_y", 7},
+    {"rear_left_knee", 8},    {"rear_right_hip_x", 9}, {"rear_right_hip_y", 10}, {"rear_right_knee", 11},
+    {"arm_sh0", 12},          {"arm_sh1", 13},         {"arm_el0", 14},          {"arm_el1", 15},
+    {"arm_wr0", 16},          {"arm_wr1", 17},         {"arm_f1x", 18},
+};
+/// @brief Maps joint name to joint index for robots without arms.
+static const std::unordered_map<std::string, size_t> kJointNameToIndexWithoutArm{
+    {"front_left_hip_x", 0},  {"front_left_hip_y", 1}, {"front_left_knee", 2},   {"front_right_hip_x", 3},
+    {"front_right_hip_y", 4}, {"front_right_knee", 5}, {"rear_left_hip_x", 6},   {"rear_left_hip_y", 7},
+    {"rear_left_knee", 8},    {"rear_right_hip_x", 9}, {"rear_right_hip_y", 10}, {"rear_right_knee", 11},
+};
+
+// TODO(tcappellari): Find a cleaner + better way to load and change these
+// kp and kd gains for a robot without an arm
+static const std::vector<float> no_arm_kp = {624, 936, 286, 624, 936, 286, 624, 936, 286, 624, 936, 286};
+static const std::vector<float> no_arm_kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04};
+// kp and kd gains for a robot with an arm
+static const std::vector<float> arm_kp = {624, 936, 286,  624, 936, 286, 624, 936, 286, 624,
+                                          936, 286, 1020, 255, 204, 102, 102, 102, 16.0};
+static const std::vector<float> arm_kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20,
+                                          5.20, 2.04, 10.2, 15.3, 10.2, 2.04, 2.04, 2.04, 0.32};
+/// @brief Given a list of joints from a JointStates message, put them in the correct order that the Spot Hardware
+/// interface expects.
+/// @param msg JointStates message
+/// @param ordered_joint_angles_ Joint positions from the joint state message following the correct order.
+/// @return boolean indicating if the joint angles got ordered successfully.
+bool order_joints(const sensor_msgs::msg::JointState& msg, std::vector<double>& ordered_joint_angles_) {
+  const auto njoints = msg.position.size();
+  ordered_joint_angles_.resize(njoints);
+  static const std::unordered_map<std::string, size_t> kJointNameToIndex;
+  // Different joint index maps for arm-full and arm-less
+  switch (njoints) {
+    // case without arm
+    case kNjointsNoArm:
+      for (size_t i = 0; i < njoints; i++) {
+        // get the joint name
+        const auto joint_name = msg.name.at(i);
+        try {
+          const auto joint_index = kJointNameToIndexWithoutArm.at(joint_name);
```

Comment:
@tcappellari-bdai meta: if we were decouple the name-to-index mapping selection to a function, that'd reduce the boilerplate. Same for gain selection below.

Reply:
ill move it around in a separate PR so I can get this merged in for now